src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.cpp

```diff
@@ -0,0 +1,443 @@
+                                            response.certificateDetails = details;
+                                            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                            return CHIP_NO_ERROR;
+                                        });
+    }
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleLookupRootCertificate(HandlerContext & ctx,
+                                                                 const LookupRootCertificate::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: LookupRootCertificate");
+    auto result = mDelegate.LookupRootCertByFingerprint(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(),
+                                                        req.fingerprint, [&](auto & certificate) -> CHIP_ERROR {
+                                                            LookupRootCertificateResponse::Type response;
+                                                            response.caid = certificate.caid;
+                                                            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                                            return CHIP_NO_ERROR;
+                                                        });
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleRemoveRootCertificate(HandlerContext & ctx,
+                                                                 const RemoveRootCertificate::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: RemoveRootCertificate");
+    auto result = mDelegate.RemoveRootCert(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(), req.caid);
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result == CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Success);
+    }
+    else
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleGenerateClientCsr(HandlerContext & ctx, const TLSClientCSR::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: TLSClientCSR");
+
+    VerifyOrReturn(req.nonce.size() <= 128, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
```

Comment:
Spec has 32?

Reply:
Apologies, that was later updated & the spec I worked off of had 128, will update with follow-on which has additional changes